src/magic_spell_effect.cpp

```diff
@@ -706,22 +710,18 @@ static void spell_move( const spell &sp, const Creature &caster,
         src_items.clear();
     }
 
-    // Helper function to move particular field type if corresponding target flag is enabled.
-    auto move_field = [&sp, from, to, &here]( spell_target target, field_type_id fid ) {
-        if( !sp.is_valid_effect_target( target ) ) {
-            return;
-        }
-        auto &src_field = here.field_at( from );
-        if( field_entry *entry = src_field.find_field( fid ) ) {
-            int intensity = entry->get_field_intensity();
-            here.remove_field( from, fid );
-            here.set_field_intensity( to, fid, intensity );
-        }
-    };
     // Moving fields.
-    move_field( spell_target::fire, field_type_id( "fd_fire" ) );
-    move_field( spell_target::blood, field_type_id( "fd_blood" ) );
-    move_field( spell_target::blood, field_type_id( "fd_gibs_flesh" ) );
+    if( sp.is_valid_target( spell_target::field ) ) {
+        map &here = get_map();
+        field &src_field = here.field_at( from );
+        for( const std::pair<const field_type_id, field_entry> &fd : src_field ) {
+            if( fd.first.is_valid() && !fd.first.id().is_null() ) {
+                const int intensity = fd.second.get_field_intensity();
+                here.remove_field( from, fd.first );
```

Comment:
Doesn't this modify `src_field`, rendering the iterator unusable?

Reply:
hrm, if it does that then i'll need to do 2 loops.